src/world.rs

```diff
@@ -68,7 +68,7 @@ impl World {
     }
 
     #[allow(unreachable_code)]
-    pub fn handle_player(&self, mut stream: TcpStream) -> io::Result<()> {
+    pub fn handle_player(&self, mut stream: SymmStream) -> io::Result<()> {
```

Comment:
We should change this to use generics and force the parameter to impl traits `Read + Write` so we can easily switch authenticated mode `on/off`. /cc @fenhl